Common/cpp/SharedItems/Shareables.h

```diff
@@ -239,19 +239,18 @@ class ShareableWorklet : public ShareableObject {
   }
 };
 
-class ShareableRemoteFunction
-    : public Shareable,
-      public std::enable_shared_from_this<ShareableRemoteFunction> {
+class RNFunctionRef : public Shareable,
+                      public std::enable_shared_from_this<RNFunctionRef> {
  private:
   jsi::Function function_;
   std::shared_ptr<JSRuntimeHelper> runtimeHelper_;
 
  public:
-  ShareableRemoteFunction(
+  RNFunctionRef(
```

Comment:
This class derives from `Shareable` so its name needs to start with `Shareable` according to the convention (`ShareableString`, `ShareableObject` etc.).

Also, the name shouldn't mention React Native (`RN`). Indeed, currently this supports only functions from RN runtime but here's how it looks on my branch which adds support for multiple runtimes:

```cpp
class ShareableRemoteFunction
    : public Shareable,
      public std::enable_shared_from_this<ShareableRemoteFunction> {
 private:
  jsi::Runtime *runtime_;
  jsi::Function function_;

 public:
  ShareableRemoteFunction(jsi::Runtime &rt, jsi::Function &&function)
      : Shareable(RemoteFunctionType),
        runtime_(&rt),
        function_(std::move(function)) {}
  jsi::Value toJSValue(jsi::Runtime &rt) override {
    if (&rt == runtime_) {
      return jsi::Value(rt, function_);
    } else {
#ifdef DEBUG
      return getValueUnpacker(rt).call(
          rt,
          ShareableJSRef::newHostObject(rt, shared_from_this()),
          jsi::String::createFromAscii(rt, "RemoteFunction"));
#else
      return ShareableJSRef::newHostObject(rt, shared_from_this());
#endif
    }
  }
};
```

As you can see, when `toJSValue` is called with the original runtime where the function comes from, it simply returns the function itself. This will work with all runtimes, not only with the RN runtime.

Finally, I'm not a huge fan of using shorthand `Ref` instead of `Reference`. I still don't know what's the difference between `Component` and `ComponentRef` and I want to save others from this pain.

I believe `ShareableFunction` would be the best option here. Obviously, such functions can't be called but at least they can be shared between runtimes and that's what shareables are for.

Reply:
Thanks for being so concise in this explanation. Those are indeed very fair points and I think `ShareableFunction` in this case is a good name. Therefore I think we should propagate it to JS to signify this being the same object and rename it from `RemoteFunction` to `ShareableFunction`. I wonder what @piaskowyk thinks about it.

Reply:
I understand the motivation of this PR. I am glad that you want to improve naming, because this is useful and important, but in this case, I am not a big fan of these changes 😅 

Personally, for me `ShareableRemoteFunction` means exactly what it is. Especially that `RNFunctionRef` is not true, because even now we can meet more than just 2 JS runtimes in the application - for example, if someone uses `rn-camera-vision`.

But overall I am open to discussion 😀

Reply:
I was always wondering what does Remote mean in this context. The name suggests that RemoteFunction is a very specific being but nowhere in the code we have it defined/described and in my opinion it turns out much more magical thanks to its name than it really is.